pennylane/ops/qubit/qchem_ops.py

```diff
@@ -464,3 +464,146 @@ def _matrix(cls, *params):
+
+    with the same orbital operation applied in the :math:`\alpha` and :math:`\beta` spin orbitals.
+
+    **Details:**
+
+    * Number of wires: 4
+    * Number of parameters: 1
+    * Gradient recipe: The ``OrbitalRotation`` operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://arxiv.org/abs/2104.05695)
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int]): the wires the operation acts on
+
+    **Example**
+
+    .. code-block::
+
+        >>> dev = qml.device('default.qubit', wires=4)
+
+        >>> @qml.qnode(dev)
+        ... def circuit(phi):
+        ...     qml.BasisState([1, 1, 0, 0], wires=[0, 1, 2, 3])
+        ...     qml.OrbitalRotation(phi, wires=[0, 1, 2, 3])
+        ...     return qml.state()
+
+        >>> circuit(0.1)
+        array([ 0.        +0.j,  0.        +0.j,  0.        +0.j,
+                0.00249792+0.j,  0.        +0.j,  0.        +0.j,
+               -0.04991671+0.j,  0.        +0.j,  0.        +0.j,
+               -0.04991671+0.j,  0.        +0.j,  0.        +0.j,
+                0.99750208+0.j,  0.        +0.j,  0.        +0.j,
+                0.        +0.j])
+    """
+
+    num_params = 1
+    num_wires = 4
+    par_domain = "R"
+    grad_method = "A"
+    grad_recipe = four_term_grad_recipe
+    generator = [
+        qml.math.array(
+            [
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, -1j, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, -1j, 0, 0, -1j, 0, 0, 0, 0, 0, 0],
+                [0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0],
+                [0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0],
+                [0, 0, 0, 0, 0, 0, 1j, 0, 0, 1j, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1j, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            ]
+        ),
+        -1 / 2,
+    ]
+
+    @classmethod
+    def _matrix(cls, *params):
+        # This matrix is the "sign flipped" version of that on p18 of https://arxiv.org/abs/2104.05695,
+        # where the sign flip is to adjust for the opposite convention used by authors for naming wires.
+        # Additionally, there was a typo in the sign of a matrix element "s" at [2, 8], which is fixed here.
+
+        phi = params[0]
+        c = qml.math.cos(phi / 2)
```

Comment:
Great to see `qml.math` in action!

Reply:
Same! it is really nice not needing to modify `tf_ops.py` etc.